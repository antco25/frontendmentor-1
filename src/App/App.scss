@import '../Components/Utils/common';

.app {
    display: flex;
    overflow: visible;
    height: 100%;
    align-items: center;
    justify-content: center;
    background-image: url('bg-intro-desktop.png');
    background-color: $primary-1;
    background-attachment: fixed;
}

.app .app-wrapper {
    display: flex;
    flex-direction: row;
    width: 75rem;
    padding: 2rem;

    section {
        flex: 1;
        margin: 1rem;

        h1 {
            font-size: 3rem;
            line-height: 1.2;
            margin: 1.5rem 0;
        }
    }

    .app-description {
        display: flex;
        align-items: center;
    }

    .app-form {
        display: flex;
        flex-direction: column;
        justify-content: center;

        .form-header {
            margin: 0;
        }
    }
}

//1100px - 992px
@media only screen and (max-width: $breakpoint-1) { 

    .app {
        min-height: 700px;
    }
    .app .app-wrapper .app-description article {
        h1 {
            font-size: $s-44;
        }
    }
    
    .app .app-wrapper .app-form {
        .card {
            padding: 1.2rem $s-16;
        } 
    }

}

//992px-768px
@media only screen and (max-width: $breakpoint-2) { 

    .app .app-wrapper {
        padding: 1rem;

        section {
            margin: $s-12;
        }
    }

    .app .app-wrapper .app-description article {
        h1 {
            font-size: $s-32;
        }

        p {
            font-size: $s-14;
        }
    } 

    .app .app-wrapper .app-form {
        .card {
            padding: $s-16 $s-10;
        }

        .form-header {
            font-size: $s-14;
            padding: 0 $s-10;
        }

        .form form {
            padding: 1rem $s-10;

            .input-wrapper {
                padding: $s-10 $s-16;
            }

            input[type=text], input[type=password] {
                font-size: $s-12;
            }

            input[type=submit] {
                font-size: $s-16;
                padding: $s-10 $s-16;
            }
        }
    }
}

//768px-576px
@media only screen and (max-width: $breakpoint-3) { 

    .app {
        min-height: 600px;
    }
    .app .app-wrapper {
        section {
            margin: $s-8;
        }
    }

    .app .app-wrapper .app-description article {
        h1 {
            font-size: $s-24;
        }

        p {
            font-size: $s-12;
        }
    } 

    .app .app-wrapper .app-form {
        .card {
            padding: $s-16 $s-10;
        }

        .form-header {
            font-size: $s-12;
            padding: 0 $s-6;
        }

        .form form {
            padding: $s-10 $s-6;

            .input-wrapper {
                padding: $s-8 $s-12;
            }

            input[type=text], input[type=password] {
                font-size: $s-10;
            }

            input[type=submit] {
                font-size: $s-14;
                padding: $s-8 $s-12;
            }

            .input p {
                font-size: $s-10;
            }

            .input .input-error {

                input[type=text], input[type=password] {
                    //Flex won't resize the input if I don't add a width
                    width: 1rem; 
                }

                img {
                    max-width: $s-18;
                }
            } 

            footer p {
                font-size: $s-10;
            }
        }
    }

}

//576px-375px
@media only screen and (max-width: $breakpoint-4) { 

    .app {
        min-height: 500px;
    }
    
    .app .app-wrapper {
        padding: $s-14;

        section {
            margin: $s-4;
        }
    }

    .app .app-wrapper .app-description article {
        h1 {
            font-size: $s-16;
        }

        p {
            font-size: $s-10;
        }
    } 

    .app .app-wrapper .app-form {
        .card {
            padding: $s-10 $s-8;
            margin-bottom: $s-24;
        }

        .form-header {
            font-size: $s-10;
        }

        .form form {
            padding: $s-10 $s-6;

            .input-wrapper {
                padding: $s-8 $s-6;
            }

            input[type=text], input[type=password] {
                font-size: $s-8;
                //Flex won't resize the input if I don't add a width
                width: 1rem; 
            }

            input[type=submit] {
                font-size: $s-10;
                padding: $s-8;
            }

            .input p {
                font-size: $s-8;
            }

            .input .input-error img {
                max-width: $s-14;
            }

            footer p {
                font-size: $s-8;
            }
        }
    }

}

//375 and smaller
@media only screen and (max-width: $breakpoint-mobile) {

    .app {
        background-image: url('bg-intro-mobile.png');
        min-height: 900px;
        min-width: 330px;
    }

    .app .app-wrapper {
        flex-direction: column;
        padding: $s-14;

        section {
            margin-bottom: $s-18;
        }
    }

    .app .app-wrapper .app-description article {
        text-align: center;

        h1 {
            font-size: $s-28;
        }

        p {
            font-size: $s-14;
        }
    } 

    .app .app-wrapper .app-form {
        .card {
            padding: $s-20 $s-12;
            margin-bottom: $s-24;
        }

        .form-header {
            font-size: $s-14;
            padding: 0 $s-48;
        }

        .form form {
            padding: $s-10 $s-6;

            .input-wrapper {
                padding: $s-10 $s-16;
            }

            input[type=text], input[type=password] {
                font-size: $s-12;
                //Flex won't resize the input if I don't add a width
                width: 1rem; 
            }

            input[type=submit] {
                font-size: $s-16;
                padding: $s-10 $s-16;
            }

            .input p {
                font-size: $s-12;
            }

            .input .input-error img {
                max-width: unset;
            }

            footer p {
                font-size: $s-12;
            }
        }
    }
}



