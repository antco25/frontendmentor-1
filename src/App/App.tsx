import './App.scss';
import Card from '../Components/Card/Card'
import Form from '../Components/Form/Form'

function App() {

  return (
    <div className="app">
      <div className="app-wrapper">
      <section className="app-description">
        <article>
        <h1>Learn to code by watching others</h1>
        <p>
          See how experienced developers solve problems in real time. Watching
          scripted tutorials is great, but understanding how developers think is invaluable.
        </p>
        </article>
      </section>
      <section className="app-form">
        <Card type={"accent"}>
          <p className="form-header"><b>Try it for free for 7 days</b> then $20/mo. thereafter</p>
        </Card>
        <Form/>
      </section>
      </div>
    </div>
  );
}


export default App;
