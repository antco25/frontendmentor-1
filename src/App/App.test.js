import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react';
import App from './App';

it('renders correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders header', () => {
  render(<App />);
  const headerText = screen.getByText(/Learn to code by watching others/);
  expect(headerText).toBeInTheDocument();
});
