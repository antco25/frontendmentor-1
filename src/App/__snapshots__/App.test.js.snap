// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="app"
>
  <div
    className="app-wrapper"
  >
    <section
      className="app-description"
    >
      <article>
        <h1>
          Learn to code by watching others
        </h1>
        <p>
          See how experienced developers solve problems in real time. Watching scripted tutorials is great, but understanding how developers think is invaluable.
        </p>
      </article>
    </section>
    <section
      className="app-form"
    >
      <div
        className="card card-accent"
      >
        <p
          className="form-header"
        >
          <b>
            Try it for free for 7 days
          </b>
           then $20/mo. thereafter
        </p>
      </div>
      <div
        className="form"
      >
        <div
          className="card card-default"
        >
          <form
            onSubmit={[Function]}
          >
            <div
              className="input"
            >
              <div
                className="input-wrapper"
              >
                <input
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="First Name"
                  type="text"
                />
              </div>
            </div>
            <div
              className="input"
            >
              <div
                className="input-wrapper"
              >
                <input
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="Last Name"
                  type="text"
                />
              </div>
            </div>
            <div
              className="input"
            >
              <div
                className="input-wrapper"
              >
                <input
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="Email Address"
                  type="text"
                />
              </div>
            </div>
            <div
              className="input"
            >
              <div
                className="input-wrapper"
              >
                <input
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="Password"
                  type="password"
                />
              </div>
            </div>
            <div
              className="input"
            >
              <input
                type="submit"
                value="Claim your free trial"
              />
            </div>
            <footer>
              <p>
                By clicking the button, you are agreeing to our 
                <a
                  href="#"
                >
                  Terms and Services
                </a>
              </p>
            </footer>
          </form>
        </div>
      </div>
    </section>
  </div>
</div>
`;
